core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -49,7 +52,8 @@ private[spark] class LiveListenerBus extends SparkListenerBus with Logging {
         // Atomically remove and process this event
         LiveListenerBus.this.synchronized {
           val event = eventQueue.poll
-          if (event == SparkListenerShutdown) {
+          if (event == null) {
+            assert(stopped)
```

Comment:
can you add an assertion message here so we get a slightly more descriptive feedback in case something goes wrong?